Oscilloscope: App crashes as soon as record button is clicked while using built-in mic.

**Actual Behaviour**
Oscilloscope: The app crashes as soon as the record button is clicked while using the built-in mic.

**Expected Behaviour**
The shouldn't crash when the record button is clicked, and I should be able to obtain the .csv files after recording.

**Steps to reproduce it**
• Open the app & click on the Oscilloscope.
• Switch to built-in mic & click on record button.
• The app crashes.

**Screenshots of the issue**
![15-12-2019_08-39-19_trim ConvGif](https://user-images.githubusercontent.com/44530973/70857556-aa973800-1f16-11ea-9ee0-a1d848b75b21.gif)

Device used - Lenovo Phab 2
Android Version - 6.0

**Would you like to work on the issue?**
No.

The Issue is there.
Clicking on use in-built mic and then start recording.
**Permission for mic given to the app**

**LogCat**
`E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
    Process: io.pslab, PID: 8223
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$3.done(AsyncTask.java:318)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:354)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:223)
        at java.util.concurrent.FutureTask.run(FutureTask.java:242)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:760)
     Caused by: java.lang.NoSuchMethodError: No static method join(Ljava/lang/CharSequence;[Ljava/lang/CharSequence;)Ljava/lang/String; in class Ljava/lang/String; or its super classes (declaration of 'java.lang.String' appears in /system/framework/core-oj.jar)
        at io.pslab.activity.OscilloscopeActivity$CaptureTask.doInBackground(OscilloscopeActivity.java:1120)
        at io.pslab.activity.OscilloscopeActivity$CaptureTask.doInBackground(OscilloscopeActivity.java:960)
        at android.os.AsyncTask$2.call(AsyncTask.java:304)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
        at java.lang.Thread.run(Thread.java:760) `

--- a/app/src/main/java/io/pslab/activity/OscilloscopeActivity.java
+++ b/app/src/main/java/io/pslab/activity/OscilloscopeActivity.java
@@ -45,6 +45,7 @@ import com.github.mikephil.charting.data.LineData;
 import com.github.mikephil.charting.data.LineDataSet;
 import com.github.mikephil.charting.interfaces.datasets.ILineDataSet;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.math3.complex.Complex;
 
 import java.util.ArrayList;
@@ -978,7 +979,7 @@ public class OscilloscopeActivity extends AppCompatActivity implements View.OnCl
                 String[] xDataString = null;
                 maxAmp = 0;
                 scienceLab.captureTraces(4, samples, timeGap, channel, isTriggerSelected, null);
-                Thread.sleep((long)(samples*timeGap*1e-3));
+                Thread.sleep((long) (samples * timeGap * 1e-3));
                 for (int i = 0; i < noOfChannels; i++) {
                     entries.add(new ArrayList<>());
                     channel = channels[i];
@@ -1052,12 +1053,12 @@ public class OscilloscopeActivity extends AppCompatActivity implements View.OnCl
                         double max = xData[xData.length - 1];
                         for (int j = 0; j < 500; j++) {
                             double x = j * max / 500;
-                            double t = 2*Math.PI*freq*(x - phase);
+                            double t = 2 * Math.PI * freq * (x - phase);
                             double y;
-                            if (t%(2*Math.PI) < 2*Math.PI*dc) {
+                            if (t % (2 * Math.PI) < 2 * Math.PI * dc) {
                                 y = offset + amp;
                             } else {
-                                y = offset - 2*amp;
+                                y = offset - 2 * amp;
                             }
                             curveFitEntries.get(curveFitEntries.size() - 1).add(new Entry((float) x, (float) y));
                         }
@@ -1097,7 +1098,7 @@ public class OscilloscopeActivity extends AppCompatActivity implements View.OnCl
                         float audioValue = (float) map(buffer[i], -32768, 32767, -3, 3);
                         if (!isFourierTransformSelected) {
                             if (noOfChannels == 1) {
-                                xDataString[i] = String.valueOf(2.0*i);
+                                xDataString[i] = String.valueOf(2.0 * i);
                             }
                             entries.get(entries.size() - 1).add(new Entry(i, audioValue));
                         } else {
@@ -1117,9 +1118,9 @@ public class OscilloscopeActivity extends AppCompatActivity implements View.OnCl
 
                 }
                 if (isRecording) {
-                    loggingXdata = String.join(" ", xDataString);
+                    loggingXdata = StringUtils.join(" ", xDataString);
                     for (int i = 0; i < yDataString.size(); i++) {
-                        loggingYdata[i] = String.join(" ", yDataString.get(i));
+                        loggingYdata[i] = StringUtils.join(" ", yDataString.get(i));
                     }
                     runOnUiThread(new Runnable() {
                         @Override

